### PR TITLE
feat: add regex pattern for HEX value in design system schema

### LINKED ISSUE #1945: Add regex pattern to HEX values in the design system schema
The design system schema, which contains HEX values, this should be included as the JSON schema [pattern](https://json-schema.org/understanding-json-schema/reference/regular_expressions.html) property.

The design system schema should exist in the `@microsoft/fast-components-styles-msft` package.

### PR DESCRIPTION
**Pull request by khamudom**
# Description

Added HEX value `pattern` property to schema for `backgroundColor` and `accentBaseColor`

closes #1945 

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components-styles-msft/src/design-system/design-system.schema.ts [hunk 1]
```diff
@@ -231,5 +232,23 @@ export default {
             type: "number",
             default: 18,
         },
+        neutralPalette: {
```
Comment (nicholasrice) [inline packages/fast-components-styles-msft/src/design-system/design-system.schema.ts]: `neutralPalette` and `accentPalette` are arrays of strings - not strings themselves.

Comment (khamudom) [reply packages/fast-components-styles-msft/src/design-system/design-system.schema.ts]: `neutralPalette` and `accentPalette` have been omitted

**Comment by janechu** [comment]
Please use the github keywords to close the issue this is resolving, see the prompt in the PR template and https://help.github.com/en/articles/closing-issues-using-keywords

**Comment by janechu** [comment]
Please tick anything in the process and policy checklist, at the very least "I have tested my changes." and "I have read the CONTRIBUTING documentation and followed the standards for this project." should be addressed, please make sure those are followed and mark them as checked when done.

**Comment by janechu** [comment]
Your failing tests can be fixed by running `lerna run prettier`

**Pull request merged**